core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +64,23 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private[spark] lazy val parallelPartitionEval: Boolean = {
```

Comment:
Then I think have `spark.rdd.enableParallelListing` default to `true`, and use the default thread pool is a good idea. This requires the manual tuning compared with  `spark.rdd.parallelListingThreshold`, and it's flexible enough to disable it. My 0.02 cents. Thanks.